dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationControllerTest.java

```diff
@@ -177,25 +180,69 @@ void shouldGetNonPaginatedItemsWithPagingSetToFalse() {
   }
 
   @Test
-  void shouldThrowPostPotentialDuplicateWhenMissingDuplicateTeiInPayload() throws Exception {
+  void shouldFailWhenSkipPagingAndPagingAreFalse() {
+    String message =
+        GET("/potentialDuplicates?paging=false&skipPaging=false")
+            .content(HttpStatus.BAD_REQUEST)
+            .getString("message")
+            .string();
+
+    assertStartsWith("Paging can either be enabled or disabled", message);
+  }
+
+  @Test
+  void shouldFailWhenSkipPagingAndPagingAreTrue() {
+    String message =
+        GET("/potentialDuplicates?paging=true&skipPaging=true")
+            .content(HttpStatus.BAD_REQUEST)
+            .getString("message")
+            .string();
+
+    assertStartsWith("Paging can either be enabled or disabled", message);
+  }
+
+  @Test
+  void shouldThrowBadRequestExceptionWhenMissingDuplicateTeInPayload() throws Exception {
     PotentialDuplicate potentialDuplicate = new PotentialDuplicate(UID.of(origin), null);
     assertStatus(
         HttpStatus.BAD_REQUEST,
         POST(ENDPOINT, objectMapper.writeValueAsString(potentialDuplicate)));
   }
 
   @Test
-  void shouldThrowPostPotentialDuplicateWhenMissingOriginTeiInPayload() throws Exception {
-    PotentialDuplicate potentialDuplicate = new PotentialDuplicate(null, UID.of(duplicate1));
+  void shouldThrowNotFoundExceptionWhenTeInPayloadDoesNotExist() throws Exception {
+    String nonExistentUID = CodeGenerator.generateUid();
+    PotentialDuplicate potentialDuplicate =
+        new PotentialDuplicate(UID.of(origin), UID.of(nonExistentUID));
+    assertStatus(
+        HttpStatus.NOT_FOUND, POST(ENDPOINT, objectMapper.writeValueAsString(potentialDuplicate)));
```

Comment:
Is there a pro of using this style vs the `POST().content(HttpStatus.OK);` that is also in our code? I find the latter more readable but if the one you are using has an advantage I can be convinced 😁

Reply:
not that I know, just used it because every other tests uses that style too, but now that you bring it up, I prefer your proposal too